### ISSUE TITLE
Unable to build on macOS 14

### ISSUE THREAD

**Issue by mangerlahn** [Project-OSRM/osrm-backend#6707]
Building this project on macOS 14 currently fails. The accompanying Xcode 15 supports a version of C++ that does not contain `std::unary_function`, a function used by boost 1.7x. This has been resolved with boost 1.8x, which can't be used here because it also breaks the build. 

The fix suggested [here](https://github.com/facebook/react-native/issues/37748) worked, I just needed to add
```
add_compile_definitions(_LIBCPP_ENABLE_CXX17_REMOVED_UNARY_BINARY_FUNCTION)
```
to the `CMakeLists.txt` file. Another workaround would of course be to restore compatibility with boost 1.8x.

The second issue I encountered was, that for some reason multiple unused result and unused parameter warnings popped up. As they were treated as errors, the build failed. By modifying the `CMakeLists.txt` again I was able to build successfully. 
I added
```
add_compile_options(-Wno-unused-result -Wno-unused-parameter)
```

I have little experience with all of this, that's why I post this as a bug report instead of a PR. Not sure if it would be better to fix the affected code.

**Comment by loweas** [comment]
I as well. First workaround helped (thank you!) but can't seem to work around the next one where I get unused-results.

**Comment by danpat** [comment]
I made a PR to suppress most of those problems in https://github.com/Project-OSRM/osrm-backend/pull/6709 - it's nothing fancy, just marks some unused results as ... unused, to keep the compiler happy.

**Comment by mangerlahn** [comment]
This has been resolved by #6709, thanks a lot!
